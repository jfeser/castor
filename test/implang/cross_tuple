AList(r1, ATuple([AScalar(0.f), AScalar((0.g - 0.f) as d)], cross))