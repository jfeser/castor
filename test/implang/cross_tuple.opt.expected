// Locals:
// i1 : int (persists=true)
// cstart4 : int (persists=true)
// cstart3 : int (persists=true)
// cstart0 : int (persists=true)
// count2 : int (persists=true)
fun printer () : void {
    cstart0 = 0;
    i1 = 0;
    count2 = 5;
    loop (i1 < count2) {
        cstart3 = cstart0;
        cstart4 = cstart3 + 1;
        print(tuple[int, int], (buf[cstart3 : 1], buf[cstart4 : 1]));
        cstart0 = cstart0 + 2;
        i1 = i1 + 1;
    }
}
// Locals:
// i1 : int (persists=true)
// cstart4 : int (persists=true)
// cstart3 : int (persists=true)
// cstart0 : int (persists=true)
// count2 : int (persists=true)
fun consumer () : void {
    cstart0 = 0;
    i1 = 0;
    count2 = 5;
    loop (i1 < count2) {
        cstart3 = cstart0;
        cstart4 = cstart3 + 1;
        consume(tuple[int, int], (buf[cstart3 : 1], buf[cstart4 : 1]));
        cstart0 = cstart0 + 2;
        i1 = i1 + 1;
    }
}
